### ISSUE TITLE
Support extensions in graphql-ws subscribe message not only for Apollo Persistent Query

### ISSUE THREAD

**Issue by russellyou** [apollographql/apollo-kotlin#4575]
### Use case

At the moment, apollo persistent query can be added to graphql-ws subscribe msg 'extensions' field.

 
```kotlin
interface SubscribeMessage {
  id: '<unique-operation-id>';
  type: 'subscribe';
  payload: {
    operationName?: string | null;
    query: string;
    variables?: Record<string, unknown> | null;
    extensions?: Record<string, unknown> | null;
  };
}
```

Could we add arbitrary extensions map?

For ex, add a subscribeExtensions parameter to GraphQLWsProtocol.Factory?  

```kotlin
class Factory(
      private val connectionPayload: suspend () -> Map<String, Any?>? = { null },
      private val pingIntervalMillis: Long = -1,
      private val pingPayload: Map<String, Any?>? = null,
      private val pongPayload: Map<String, Any?>? = null,
      private val connectionAcknowledgeTimeoutMs: Long = 10_000,
      private val frameType: WsFrameType = WsFrameType.Text,
  )
```

**Comment by martinbonnin** [comment]
Hi 👋 

Are you trying to get [APQ](https://www.apollographql.com/docs/apollo-server/performance/apq/) working over `graphql-ws`? If yes, I think you should be able to enable APQs like with HTTP because it uses the [same underlying code](https://github.com/apollographql/apollo-kotlin/blob/75242318c9c9c6db4313175a90c94729681d2f1b/libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/network/ws/protocol/GraphQLWsProtocol.java#L107). You can enable APQs when building your client:

```kotlin
    val apolloClient = ApolloClient.Builder()
            .serverUrl(serverUrl)
            .autoPersistedQueries()
            .build()
``` 

I think that should send payloads like this:

```json
{
  "query": "...",
  "extensions": {
    "persistedQuery": {
      "version": 1,
      "sha256Hash": "ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"
    }
  }
}
```

We don't have any tests for this so definitely let us know if this doesn't work!

**Comment by russellyou** [comment]
Thanks . No I don't want APQ. I just want to add arbitrary extensions in graphq-ws subscribe msg. Is it possible
``` {
  id: '<unique-operation-id>',
  type: 'subscribe',
  payload: {
    query: "gql query";
    extensions: {
    "key1": {
      "key2": "value2"
    }
  }
  }
}

**Comment by martinbonnin** [comment]
Gotcha 👍 

I think we'll want the `requestComposer` to be customizable for WebSockets like it is for HTTP. There's nothing like that just yet though so your best bet is to copy paste [GraphQLWSProtocol](https://github.com/apollographql/apollo-kotlin/blob/80e763301181a0c2d342be2a329c4be4ce720c5a/libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/ws/GraphQLWsProtocol.kt#L18) and customize it to your needs.

**Comment by martinbonnin** [comment]
Howdy 👋 https://github.com/apollographql/apollo-kotlin/pull/4589 introduces `WebSocketPayloadComposer` for your payload customization needs. It's all `@ApolloExperimental` and will be available in next version (or in a SNAPSHOT once this PR merged)

**Comment by BoD** [comment]
This is now available in v3.7.3.
